### PR TITLE
UIUC API Fuzzer updated to env

### PR DESCRIPTION
**Pull request by akshathsk**

### CODE REVIEW

File: uiuc_tool_tester.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        enable_gpt_logs = sys.argv[3]
+        if enable_gpt_logs.lower() == 'true':
+            enable_gpt_logs = True
+    except:
+        enable_gpt_logs = False
+        
+    runs = sys.argv[4]
+
+    curdir = os.getcwd()
+    # services = ["market","user-management", "cwa-verification","ncs", "proxyprint", "restcountries", "scout-api", "erc20-rest-service", "person-controller", "rest-study", "spring-batch-rest", "project-tracking-system"]
```
Comment (akshathsk) [inline uiuc_tool_tester.py:35]: Remove comments

File: UIUC-API-Tester/APITester/uiuc_api_tester.py [hunk 2]
```diff
@@ -0,0 +1,1083 @@
+import openai
```
Comment (akshathsk) [inline UIUC-API-Tester/APITester/uiuc_api_tester.py:1]: Format the file and remove unused imports

File: tool_report.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+
+    today = datetime.date.today()
+    path = '/home/darko/api-tester/REST_Go/'+str(tool)+'_results_'+str(today)+'.xlsx'
+    writer = pd.ExcelWriter(path, engine = 'openpyxl')
+    final_df_coverage.to_excel(writer,sheet_name='coverage')
+    final_df_error.to_excel(writer, sheet_name='bugs')
+    writer.close()
+
+
```
Comment (akshathsk) [inline tool_report.py:86]: Remove additional lines of code

File: run_tool.py [hunk 4]
```diff
@@ -49,12 +52,24 @@ def blackbox(swagger, port):
...
+            subprocess.run("cd UIUC-API-Tester/APITester && python3 integrate_enum.py", shell=True)
+            subprocess.run("cd UIUC-API-Tester/APITester && python3 uiuc_api_tester.py > uiuc_test_6500.txt", shell=True)
+            break
 
 if __name__ == "__main__":
     tool = sys.argv[1]
     service = sys.argv[2]
     port = sys.argv[3]
     time_limit = "1"
+    # print(service)
```
Comment (akshathsk) [inline run_tool.py:66]: remove this comment

File: run_modified.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import sys
```
Comment (akshathsk) [inline run_modified.py:1]: Remove this file as it is not being used anymore

File: run_instructions.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
+Inorder to execute the RESTGPT tool , the following scripts need to be present in the project ( starting from root directiory RESTGO/)
```
Comment (akshathsk) [inline run_instructions.md:1]: Delete this file as it is a duplicate

File: report.py [hunk 7]
```diff
@@ -39,13 +39,15 @@
```
Comment (akshathsk) [inline report.py:39]: remove all unwanted print statements

File: report.py [hunk 8]
```diff
@@ -252,31 +267,37 @@
...
         method = 0
     res = res + str(line*100) + '%,' + str(branch*100) + '%,' + str(method*100) + '%\n'
 res = res + str(error) + ',' + str(unique_err) + ',' + str(crucial) + '\n'
 
 with open('res.csv', 'w') as f:
     f.write(res)
+    print("writing success")
 
+print(k, services[k])
+#subprocess.run("sudo mkdir -p " + "data/" + services[k], shell=True)
```
Comment (akshathsk) [inline report.py:292]: Remove these comments

File: report.py [hunk 9]
```diff
@@ -39,13 +39,15 @@
 
 port = sys.argv[1]
 name = sys.argv[2]
+tool = sys.argv[3]
```
Comment (akshathsk) [inline report.py:42]: Format the file

File: UIUC-API-Tester/input/config.json [hunk 10]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (akshathsk) [inline UIUC-API-Tester/input/config.json:1]: delete this file

File: UIUC-API-Tester/APITester/integrate_enum.py [hunk 11]
```diff
@@ -0,0 +1,62 @@
+import json
```
Comment (akshathsk) [inline UIUC-API-Tester/APITester/integrate_enum.py:1]: format this file

File: UIUC-API-Tester/APITester/codeql_enum_analysis.py [hunk 12]
```diff
@@ -0,0 +1,28 @@
+import re
```
Comment (akshathsk) [inline UIUC-API-Tester/APITester/codeql_enum_analysis.py:1]: format this file

**Comment by akshathsk** [comment]
Path seems to be hard coded in the dotnet setup in run_tool.py

```
# Add DotNet 5 Bin Path to env
    path_to_dotnet5 = '/home/darko/dotnet'
    path_var = os.environ['PATH']
    new_path = f'{path_to_dotnet5}:{path_var}'
    os.environ['PATH'] = new_path
    print(service)
```

Asked the team to raise another PR to fix the hard coding

File: uiuc_tool_tester.py (hunk 1 above)
Comment (Suraj-Vashista-BK) [reply uiuc_tool_tester.py]: Made changes as suggested.

File: UIUC-API-Tester/APITester/uiuc_api_tester.py (hunk 2 above)
Comment (Suraj-Vashista-BK) [reply UIUC-API-Tester/APITester/uiuc_api_tester.py:1]: Made changes as suggested.

File: tool_report.py (hunk 3 above)
Comment (Suraj-Vashista-BK) [reply tool_report.py]: Made changes as suggested.

File: run_tool.py (hunk 4 above)
Comment (Suraj-Vashista-BK) [reply run_tool.py:66]: Made changes as suggested.

File: run_modified.py (hunk 5 above)
Comment (Suraj-Vashista-BK) [reply run_modified.py]: Made changes as suggested.

File: run_instructions.md (hunk 6 above)
Comment (Suraj-Vashista-BK) [reply run_instructions.md]: Made changes as suggested.

File: report.py (hunk 7 above)
Comment (Suraj-Vashista-BK) [reply report.py:41]: Made changes as suggested.

File: report.py (hunk 8 above)
Comment (Suraj-Vashista-BK) [reply report.py]: Made changes as suggested.

File: report.py (hunk 9 above)
Comment (Suraj-Vashista-BK) [reply report.py:44]: Made changes as suggested.

File: UIUC-API-Tester/input/config.json (hunk 10 above)
Comment (Suraj-Vashista-BK) [reply UIUC-API-Tester/input/config.json]: Made changes as suggested.

File: UIUC-API-Tester/APITester/integrate_enum.py (hunk 11 above)
Comment (Suraj-Vashista-BK) [reply UIUC-API-Tester/APITester/integrate_enum.py:1]: Made changes as suggested.

File: UIUC-API-Tester/APITester/codeql_enum_analysis.py (hunk 12 above)
Comment (Suraj-Vashista-BK) [reply UIUC-API-Tester/APITester/codeql_enum_analysis.py:1]: Made changes as suggested.

**Comment by akshathsk** [comment]
LGTM!

**Pull request merged**